server/games/game_results.py

```diff
@@ -75,21 +75,38 @@ def outcome(self, army: int) -> GameOutcome:
         if army not in self:
             return GameOutcome.UNKNOWN
 
-        outcomes = set(
-            r.outcome for r in self[army]
-            if r.outcome is not GameOutcome.UNKNOWN
+        voters = defaultdict(set)
+        for report in filter(
+            lambda r: r.outcome is not GameOutcome.UNKNOWN, self[army]
+        ):
+            voters[report.outcome].add(report.reporter)
+
+        if len(voters) == 0:
+            return GameOutcome.UNKNOWN
+
+        if len(voters) == 1:
+            unique_outcome = next(iter(voters.keys()))
```

Comment:
`return voters.keys()[0]` is better

Reply:
That gives me
```
TypeError: 'dict_keys' object does not support indexing
```

Reply:
Heh, in python 2 it worked... Sorry my bad.

Reply:
Yea, `.keys()`, `.items()` and `.values()` return iterables because it's faster than allocating a list when you don't need it.

Reply:
In this case you could actually use `popitem` although this modifies the dictionary.
```suggestion
            unique_outcome = voters.popitem()[0]
```